### PR TITLE
Run ViewResponseFilter only if controller was executed successfully

### PR DESCRIPTION
**Pull request by chkal**
This pull requests contains the refactoring described in #104. Actually the change is quite simple. The request-scoped`RequestLifecycle` class just tracks if the controller was invoked successfully and the `VewiResponseFilter` only runs if this is the case. 

The PR contains integration tests which checks 3 different scenarios:

  * If the controller execution is successful, the request is processed just like before.
  * If the controller throws an exception and an exception mapper creates a plain text response, this must work exactly like in a plain JAX-RS application (and NOT trigger a view engine lookup)
  * If the controller throws an exception and an exception mapper wants to render an MVC view, it can simple use a `Viewable` as the entity.

I guess this should cover all the relevant cases. The integration test actually reproduces the exact case described in #34. I don't have an explicit coverage for #91 yet, but I'm pretty sure that this is fixed as well.

The PR also contains another fix: The `AroundControllerInterceptor` wasn't detecting PATCH methods and was not implementing the MVC / JAX-RS annotation inheritance rules correctly. This caused test failures after my refactoring. The fix was easy, because there is already a utility method which checks if a method is a controller method.

Let me know what you think.

### CODE REVIEW

**Comment by erdlet** [comment]
```
Tests in error: 

  AnnotationsIT.testNoOverrideMvc:80 NullPointer
```

Only one error, but it occured also in #94.

**Comment by chkal** [comment]
> Even tough the build fails, I approve the PR because the build fail is in my opinion nothing which is directly caused by this change.

Unfortunately I'm pretty sure that these failures reveal some hidden bug in Krazo. Because with my patch the view rendering only works if the CDI interceptors are applied correctly. And it looks like that this isn't the case for these weird corner cases related to annotation inheritance. :disappointed: 

> I think you'd need to add me to the project as contributor or something similar. Now I can approve also, but it isn't counted into the PR checks. But that's OK for me.

Unfortunately we cannot change this directly, because we don't have admin permissions for the repo. However, of course you are always welcome to review ANY pull request. :+1:

**Comment by gtudan** [comment]
> Unfortunately I'm pretty sure that these failures reveal some hidden bug in Krazo. Because with my patch the view rendering only works if the CDI interceptors are applied correctly. And it looks like that this isn't the case for these weird corner cases related to annotation inheritance.

That rings a bell... remember #29? I think I saw something similar there.

**Comment by chkal** [comment]
> That rings a bell... remember #29? I think I saw something similar there.

Yeah, possible. The inheritance rules are tricky to implement.

**Comment by chkal** [comment]
Looks like all the tests are green now. There were two major issues which caused quite some pain:
  * There was a corner case in the annotation inheritance implementation which caused one test to fail. See af64422eb7017dd0b9a59a1beeade321a3e381dc for details.
  * A few of the old tests checked that `@View` is considered if a controller methods fails with an exception. This doesn't make much sense to me, because this `@View` would also be used for the success case. I adjusted these old tests in 450657d68151423e1a5aa8705d7846d8b6c979b0 and 61cb5615f7bc9a1c6f879ab108af999de55f2103

File: core/src/main/java/org/eclipse/krazo/util/ControllerUtils.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ static boolean isRequestMethod(Method method) {
             return true;
         }
         // inheritance disabled if other MVC or JAX-RS annotations found
```
Comment (erdlet) [inline core/src/main/java/org/eclipse/krazo/util/ControllerUtils.java]: We should remove this comment as it is wrong now. The method name speaks for itself here.

Comment (chkal) [reply core/src/main/java/org/eclipse/krazo/util/ControllerUtils.java]: Good catch. I adjusted the comment.
